openassessment/xblock/static/js/src/lms/oa_base.js

```diff
@@ -68,6 +69,67 @@ OpenAssessment.BaseView.prototype = {
     },
 
     /**
+     * Checks the rendering status of the views.
+     * @returns {boolean} true if any step's view is still loading.
+     */
+    isLoading: function () {
+        return this.responseView.isRendering ||
+            this.peerView.isRendering ||
+            this.selfView.isRendering ||
+            this.gradeView.isRendering ||
+            this.trainingView.isRendering ||
+            this.staffView.isRendering;
+    },
+
+    /**
+     * Adds the text to the Aria live region if included and all views are rendered.
+     * @param text
+     */
+    announceStatusChangeToSR: function(text){
+
+        if (text !== undefined) {
+            this.srStatusUpdates.push(text);
+        }
+        if (!this.isLoading() && this.srStatusUpdates.length > 0){
+            window.SR.readTexts(this.srStatusUpdates);
+            this.srStatusUpdates = []
+        }
+    },
+
+    /**
+     * Retrieves and returns the current status of a given step.
+     *
+     * @param stepID - The id of the Step to retrieve status for.
+     * @param usageID - The Usage ID of the current status.
+     * @returns {string} - the current assessment status.
+     */
+    getAssessmentStatus: function(stepID, usageID) {
+
+        var block_id = "[data-usage-id='" + usageID + "'] ";
+        var css_base = block_id + stepID + " .step__header .step__title ";
+        var css_string_title = css_base + ".step__label";
+        var css_string_status = css_base + ".step__status";
+
+        return $(css_string_title).text().trim() + ' ' + $(css_string_status).text().trim();
+    },
+
+    /**
+     * Retrieves and returns the current status of a grade step.
+     *
+     * @param stepID - The id of the Step to retrieve grade status for.
+     * @param usageID - The Usage ID of the current grade status.
+     * @returns {string} - the current grade status.
+     */
+    getAssessmentGradeStatus: function(stepID, usageID) {
+        var block_id = "[data-usage-id='" + usageID + "'] ";
+        var css_base = block_id + stepID + " .step__header .step__title ";
+        var css_string_title = css_base + ".step__label";
+        var css_string_status = css_base + ".grade__value";
```

Comment:
It looks like this line is the only difference between this method and getAssessmentStatus. Can refactor and share code.